src/dashboard/src/components/backlog/views.py

```diff
@@ -0,0 +1,218 @@
+    """
+    conn = elasticSearchFunctions.connect_and_create_index('transfers')
+
+    query = {
+        'query': {
+            'bool': {
+                'must': {
+                    'match': {
+                        'pending_deletion': True
+                    }
+                }
+            }
+        }
+    }
+
+    deletion_pending_results = conn.search(
+        body=query,
+        index='transfers',
+        doc_type='transfer',
+        fields='uuid,status'
+    )
+
+    for hit in deletion_pending_results['hits']['hits']:
+        transfer_uuid = hit['fields']['uuid'][0]
+
+        api_results = storage_service.get_file_info(uuid=transfer_uuid)
+        try:
+            status = api_results[0]['status']
+        except IndexError:
+            logger.info('Transfer not found in storage service: {}'.format(transfer_uuid))
+            continue
+
+        if status == 'DELETED':
+            elasticSearchFunctions.connect_and_remove_backlog_transfer_files(transfer_uuid)
+            elasticSearchFunctions.connect_and_remove_backlog_transfer(transfer_uuid)
+
+
+def execute(request):
+    """
+    Remove any deleted transfers from ES and render main backlog page.
+
+    :param request: The Django request object
+    :return: The main backlog page rendered
+    """
+    check_and_remove_deleted_transfers()
+    return render(request, 'backlog/backlog.html', locals())
+
+
+def get_es_property_from_column_index(index, file_mode):
+    """
+    When the user clicks a column header in the data table, we'll receive info in the ajax request
+    telling us which column # we're supposed to sort across in our query. This function will translate
+    the column index to the corresponding property name we'll tell ES to sort on.
+
+    :param index: The column index that the data table says we're sorting on
+    :param file_mode: Whether we're looking at transfers or transfer files
+    :return: The ES document property name corresponding to the column index in the data table.
+    """
+    table_columns = (
+        ('name', 'uuid', 'file_count', 'ingest_date', None),     # Transfers are being displayed
+        ('filename', 'sipuuid', None)                            # Transfer files are being displayed
+    )
+
+    if index < 0 or index >= len(table_columns[file_mode]):
+        logger.warning('Backlog column index specified is invalid for sorting, got {}'.format(index))
+        index = 0
+
+    return table_columns[file_mode][index]
+
+
+@decorators.elasticsearch_required()
+def search(request):
+    """
+    A JSON end point that returns results for various backlog transfers and their files.
+
+    :param request: The Django request object
+    :return: A JSON object including required metadata for the datatable and the backlog search results.
+    """
+    # get search parameters from request
+    queries, ops, fields, types = advanced_search.search_parameter_prep(request)
+
+    file_mode = request.GET.get('file_mode') == 'true'
```

Comment:
I use file_mode twice so I think it's best to keep it in a variable instead of using request.GET.get('file_mode') == 'true' multiple times.